File: server/data/results/translations/plain/cpp/fibonacci_memo.cpp
#include <iostream>
#include <unordered_map>

int fib(int n, std::unordered_map<int, int>& memo) {
    if (n <= 1) {
        return n;
    }
    if (memo.find(n) != memo.end()) {
        return memo[n];
    }
    int value = fib(n - 1, memo) + fib(n - 2, memo);
    memo[n] = value;
    return value;
}

int main() {
    std::unordered_map<int, int> memo;
    for (int i = 0; i < 15; ++i) {
        std::cout << fib(i, memo) << std::endl;
    }
    return 0;
}

File: server/data/results/translations/plain/go/fibonacci_memo.go
package main

import "fmt"

func fib(n int, memo map[int]int) int {
	if n <= 1 {
		return n
	}
	if value, exists := memo[n]; exists {
		return value
	}
	value := fib(n-1, memo) + fib(n-2, memo)
	memo[n] = value
	return value
}

func main() {
	memo := make(map[int]int)
	for i := 0; i < 15; i++ {
		fmt.Println(fib(i, memo))
	}
}